hcl2/transformer.py

```diff
@@ -4,7 +4,7 @@
 
 from lark import Transformer, Discard
 
-HEREDOC_PATTERN = re.compile(r'<<([a-zA-Z0-9]+)\n((.|\n)*?)\n\s*\1', re.S)
+HEREDOC_PATTERN = re.compile(r'<<([a-zA-Z][a-zA-Z0-9._-]+)\n((.|\n)*?)\n\s*\1', re.S)
```

Comment:
This could be updated to support a leading hyphen. Docs say it can start with a hyphen to indicate the content should be dedented. Though that behavior is not implemented in the heredoc parsing at the moment.

Reply:
Added support for `<<-` too